File: health_advanced/HealthDashboard/Controllers/HDExerciseSummaryViewController.m
//
//  HDExerciseSummaryViewController.m
//  HealthDashboard
//
//  Created by zhang, haizi on 2026/3/21.
//

#import "HDExerciseSummaryViewController.h"

@interface HDExerciseSummaryViewController ()
@property (nonatomic, strong) UILabel *titleLabel;
@property (nonatomic, strong) UILabel *timeLabel;
@property (nonatomic, strong) UILabel *distanceLabel;
@property (nonatomic, strong) UILabel *paceLabel;
@property (nonatomic, strong) UILabel *calorieLabel;
@property (nonatomic, strong) UIButton *saveButton;
@property (nonatomic, strong) UIButton *cancelButton;
@end

@implementation HDExerciseSummaryViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self setupUI];
    [self applyTheme];
}

- (void)setupUI {
    self.view.backgroundColor = UIColor.systemBackgroundColor;
    self.title = @"运动总结";
    
    // 标题
    _titleLabel = [UILabel new];
    _titleLabel.text = [NSString stringWithFormat:@"🎉 %@完成", self.exerciseRecord.typeString];
    _titleLabel.font = [UIFont systemFontOfSize:24 weight:UIFontWeightBold];
    _titleLabel.textColor = UIColor.labelColor;
    _titleLabel.textAlignment = NSTextAlignmentCenter;
    [self.view addSubview:_titleLabel];
    
    // 时间
    _timeLabel = [UILabel new];
    NSInteger hours = self.exerciseRecord.durationSeconds / 3600;
    NSInteger minutes = (self.exerciseRecord.durationSeconds % 3600) / 60;
    NSInteger seconds = self.exerciseRecord.durationSeconds % 60;
    _timeLabel.text = [NSString stringWithFormat:@"⏱ 时间: %02ld:%02ld:%02ld", hours, minutes, seconds];
    _timeLabel.font = [UIFont systemFontOfSize:18 weight:UIFontWeightSemibold];
    _timeLabel.textColor = UIColor.labelColor;
    [self.view addSubview:_timeLabel];
    
    // 距离
    _distanceLabel = [UILabel new];
    _distanceLabel.text = [NSString stringWithFormat:@"📍 距离: %.1f km", self.exerciseRecord.distanceKM];
    _distanceLabel.font = [UIFont systemFontOfSize:18 weight:UIFontWeightSemibold];
    _distanceLabel.textColor = UIColor.systemBlueColor;
    [self.view addSubview:_distanceLabel];
    
    // 配速
    _paceLabel = [UILabel new];
    CGFloat paceMinutes = self.exerciseRecord.durationSeconds / 60.0 / self.exerciseRecord.distanceKM;
    NSInteger paceMin = (NSInteger)paceMinutes;
    NSInteger paceSec = (NSInteger)((paceMinutes - paceMin) * 60);
    _paceLabel.text = [NSString stringWithFormat:@"🏃 配速: %ld:%02ld /km", paceMin, paceSec];
    _paceLabel.font = [UIFont systemFontOfSize:18 weight:UIFontWeightSemibold];
    _paceLabel.textColor = UIColor.labelColor;
    [self.view addSubview:_paceLabel];
    
    // 卡路里
    _calorieLabel = [UILabel new];
    _calorieLabel.text = [NSString stringWithFormat:@"🔥 卡路里: %ld kcal", self.exerciseRecord.caloriesBurned];
    _calorieLabel.font = [UIFont systemFontOfSize:18 weight:UIFontWeightSemibold];
    _calorieLabel.textColor = UIColor.systemOrangeColor;
    [self.view addSubview:_calorieLabel];
    
    // 保存按钮
    _saveButton = [UIButton buttonWithType:UIButtonTypeSystem];
    _saveButton.titleLabel.font = [UIFont systemFontOfSize:18 weight:UIFontWeightSemibold];
    [_saveButton setTitle:@"保存" forState:UIControlStateNormal];
    _saveButton.layer.cornerRadius = 8;
    _saveButton.backgroundColor = UIColor.systemGreenColor;
    [_saveButton setTitleColor:UIColor.whiteColor forState:UIControlStateNormal];
    [_saveButton addTarget:self action:@selector(saveExercise) forControlEvents:UIControlEventTouchUpInside];
    [self.view addSubview:_saveButton];
    
    // 放弃按钮
    _cancelButton = [UIButton buttonWithType:UIButtonTypeSystem];
    _cancelButton.titleLabel.font = [UIFont systemFontOfSize:18 weight:UIFontWeightSemibold];
    [_cancelButton setTitle:@"放弃" forState:UIControlStateNormal];
    _cancelButton.layer.cornerRadius = 8;
    _cancelButton.backgroundColor = UIColor.systemRedColor;
    [_cancelButton setTitleColor:UIColor.whiteColor forState:UIControlStateNormal];
    [_cancelButton addTarget:self action:@selector(cancelExercise) forControlEvents:UIControlEventTouchUpInside];
    [self.view addSubview:_cancelButton];
    
    [self layoutUI];
}

- (void)layoutUI {
    CGFloat padding = 16;
    CGFloat width = self.view.bounds.size.width - padding * 2;
    
    _titleLabel.frame = CGRectMake(padding, 100, width, 40);
    _timeLabel.frame = CGRectMake(padding, 140, width, 30);
    _distanceLabel.frame = CGRectMake(padding, 180, width, 30);
    _paceLabel.frame = CGRectMake(padding, 220, width, 30);
    _calorieLabel.frame = CGRectMake(padding, 260, width, 30);
    
    _saveButton.frame = CGRectMake(padding, self.view.bounds.size.height - padding - 120, width / 2 - 8, 50);
    _cancelButton.frame = CGRectMake(padding + width / 2 + 8, self.view.bounds.size.height - padding - 120, width / 2 - 8, 50);
}

- (void)saveExercise {
    // 数据已在计时器中保存，这里只需返回
    [self.navigationController popToRootViewControllerAnimated:YES];
}

- (void)cancelExercise {
    // 返回到类型选择页面
    [self.navigationController popToRootViewControllerAnimated:YES];
}

- (void)applyTheme {
    self.view.backgroundColor = UIColor.systemBackgroundColor;
    _titleLabel.textColor = UIColor.labelColor;
    _timeLabel.textColor = UIColor.labelColor;
    _distanceLabel.textColor = UIColor.systemBlueColor;
    _paceLabel.textColor = UIColor.labelColor;
    _calorieLabel.textColor = UIColor.systemOrangeColor;
}

@end

File: health_advanced_swift/HealthDashboard/Controllers/HDExerciseSummaryViewController.swift
//
//  HDExerciseSummaryViewController.swift
//  HealthDashboard - 运动总结
//
//  Created by AI Assistant on 2026/3/22.
//

import UIKit

// MARK: - ViewController

@MainActor
class HDExerciseSummaryViewController: UIViewController {

    // MARK: - Properties

    /// 运动记录（由计时页面传入）
    var exerciseRecord: HDExerciseRecord?

    // MARK: - UI

    private let titleLabel = UILabel()
    private let timeLabel = UILabel()
    private let distanceLabel = UILabel()
    private let paceLabel = UILabel()
    private let calorieLabel = UILabel()
    private let saveButton = UIButton(type: .custom)
    private let cancelButton = UIButton(type: .custom)

    // MARK: - Lifecycle

    override func viewDidLoad() {
        super.viewDidLoad()
        setupUI()
        applyTheme()
        NotificationCenter.default.addObserver(self, selector: #selector(applyTheme), name: NSNotification.Name("HDThemeDidChange"), object: nil)
    }

    override func viewDidLayoutSubviews() {
        super.viewDidLayoutSubviews()
        layoutUI()
    }

    // MARK: - UI Setup

    private func setupUI() {
        title = "运动总结"

        let record = exerciseRecord

        // 标题
        titleLabel.text = "\u{1F389} \(record?.typeString ?? "运动")完成"
        titleLabel.font = .systemFont(ofSize: 24, weight: .bold)
        titleLabel.textAlignment = .center
        view.addSubview(titleLabel)

        // 时间
        let totalSec = record?.durationSeconds ?? 0
        let h = totalSec / 3600
        let m = (totalSec % 3600) / 60
        let s = totalSec % 60
        timeLabel.text = String(format: "\u{23F1} 时间: %02d:%02d:%02d", h, m, s)
        timeLabel.font = .systemFont(ofSize: 18, weight: .semibold)
        view.addSubview(timeLabel)

        // 距离
        let dist = record?.distanceKM ?? 0
        distanceLabel.text = String(format: "\u{1F4CD} 距离: %.1f km", dist)
        distanceLabel.font = .systemFont(ofSize: 18, weight: .semibold)
        distanceLabel.textColor = .systemBlue
        view.addSubview(distanceLabel)

        // 配速
        var paceText = "\u{1F3C3} 配速: --"
        if dist > 0 {
            let paceMinutes = Double(totalSec) / 60.0 / Double(dist)
            let paceMin = Int(paceMinutes)
            let paceSec = Int((paceMinutes - Double(paceMin)) * 60)
            paceText = String(format: "\u{1F3C3} 配速: %d:%02d /km", paceMin, paceSec)
        }
        paceLabel.text = paceText
        paceLabel.font = .systemFont(ofSize: 18, weight: .semibold)
        view.addSubview(paceLabel)

        // 卡路里
        calorieLabel.text = "\u{1F525} 卡路里: \(record?.caloriesBurned ?? 0) kcal"
        calorieLabel.font = .systemFont(ofSize: 18, weight: .semibold)
        calorieLabel.textColor = .systemOrange
        view.addSubview(calorieLabel)

        // 保存按钮
        saveButton.setTitle("保存", for: .normal)
        saveButton.titleLabel?.font = .systemFont(ofSize: 18, weight: .semibold)
        saveButton.layer.cornerRadius = 8
        saveButton.backgroundColor = .systemGreen
        saveButton.setTitleColor(.white, for: .normal)
        saveButton.addTarget(self, action: #selector(saveExercise), for: .touchUpInside)
        view.addSubview(saveButton)

        // 放弃按钮
        cancelButton.setTitle("放弃", for: .normal)
        cancelButton.titleLabel?.font = .systemFont(ofSize: 18, weight: .semibold)
        cancelButton.layer.cornerRadius = 8
        cancelButton.backgroundColor = .systemRed
        cancelButton.setTitleColor(.white, for: .normal)
        cancelButton.addTarget(self, action: #selector(cancelExercise), for: .touchUpInside)
        view.addSubview(cancelButton)
    }

    private func layoutUI() {
        let padding: CGFloat = 16
        let width = view.bounds.width - padding * 2

        titleLabel.frame = CGRect(x: padding, y: 100, width: width, height: 40)
        timeLabel.frame = CGRect(x: padding, y: 150, width: width, height: 30)
        distanceLabel.frame = CGRect(x: padding, y: 190, width: width, height: 30)
        paceLabel.frame = CGRect(x: padding, y: 230, width: width, height: 30)
        calorieLabel.frame = CGRect(x: padding, y: 270, width: width, height: 30)

        let btnY = view.bounds.height - padding - 120
        let btnW = (width - 16) / 2
        saveButton.frame = CGRect(x: padding, y: btnY, width: btnW, height: 50)
        cancelButton.frame = CGRect(x: padding + btnW + 16, y: btnY, width: btnW, height: 50)
    }

    // MARK: - Actions

    @objc private func saveExercise() {
        // 数据已在计时页面保存到 Model，此处直接返回根页面
        navigationController?.popToRootViewController(animated: true)
    }

    @objc private func cancelExercise() {
        navigationController?.popToRootViewController(animated: true)
    }

    // MARK: - Theme

    @objc func applyTheme() {
        view.backgroundColor = .systemBackground
        titleLabel.textColor = .label
        timeLabel.textColor = .label
        distanceLabel.textColor = .systemBlue
        paceLabel.textColor = .label
        calorieLabel.textColor = .systemOrange
    }
}
